lua/neogit/lib/git/branch.lua

```diff
@@ -0,0 +1,77 @@
+local a = require('neogit.async')
+local cli = require('neogit.lib.git.cli')
+local util = require('neogit.lib.util')
+local input = require('neogit.lib.input')
+local M = {}
+
+local function contains(table, val)
+   for i=1,#table do
+      if table[i] == val then
+         return true
+      end
+   end
+   return false
+end
+
+local function parse_branches(output)
+  local branches = util.split(output, '\n')
+  local other_branches = {}
+  for _, b in ipairs(branches) do
+    local branch_name = b:match('^  (.+)')
+    if branch_name then table.insert(other_branches, branch_name) end
+  end
+
+  return other_branches
+end
+
+local get_local_branches = a.sync(function ()
+  local branches = a.wait(cli.branch
+    .list
+    .call())
+
+  return parse_branches(branches)
+end)
+
+local get_all_branches = a.sync(function ()
+  local branches = a.wait(cli.branch
+    .list
+    .all
+    .call())
+
+  return parse_branches(branches)
+end)
+
+local function prompt_for_branch(options)
+  local chosen = input.get_user_input_with_completion('branch > ', options)
+  if not contains(options, chosen) then
```

Comment:
When being inside the prompt and hitting `ESC` the input should be cancelled in my opinion. Currently I also get the error message when trying to cancel the prompt.

Reply:
Good catch, haven't even tried cancelling :X